### ISSUE TITLE
AWSS3 SSE-C stop sending data chunk for large file

### ISSUE THREAD

**Issue by brajes** [aws/aws-sdk-ios#237]
Hi,

I've been trying SSE-C with AWSS3TransferManager.
It seems to be working with file size less than 5MB.
But didn't fully working when file is larger.

The transfer stop sending after reach 686490 bytes
```
WSiOSSDKv2 [Debug] AWSSignature.m line:655 | -[AWSS3ChunkedEncodingInputStream nextChunk] | stream read: 32677, chunk size: 32768
prog 0.0327343940734863
total 686490
total 20971520
```

```
let credentialsProvider : AWSCognitoCredentialsProvider = AWSCognitoCredentialsProvider(regionType: CognitoRegionType, identityPoolId: CognitoIdentityPoolId)
let configuration : AWSServiceConfiguration = AWSServiceConfiguration(region: DefaultServiceRegionType, credentialsProvider: credentialsProvider)
AWSServiceManager.defaultServiceManager().defaultServiceConfiguration = configuration
var transferManager: AWSS3TransferManager = AWSS3TransferManager.defaultS3TransferManager()

let uploadFileURL = NSURL.fileURLWithPath(NSTemporaryDirectory() + S3UploadKeyName)
let uploadRequest = AWSS3TransferManagerUploadRequest()
uploadRequest.body = uploadFileURL
uploadRequest.key = S3UploadKeyName
uploadRequest.bucket = S3BucketName
uploadRequest.SSECustomerAlgorithm = "AES256"
uploadRequest.SSECustomerKey = "SSECustomerKey"
uploadRequest.SSECustomerKeyMD5 = "SSECustomerKeyMD5"

transferManager.upload(uploadRequest).continueWithBlock { (task) -> AnyObject! in
 //
 //
}
```

**Comment by yyzhu** [comment]
Hi @brajes 

I can confirm it is a bug in our current Mobile SDK, we will fix it in our future release.

Thanks for the reporting.

Yi

**Comment by sugampandey** [comment]
Any updates on this @yyzhu - any idea when would you guys be releasing the fix?

**Comment by yyzhu** [comment]
Hi @sugampandey @brajes 
We've just released AWS Mobile SDK for iOS 2.2.6 which resolved this issue. For more information, please see the [release notes](https://aws.amazon.com/releasenotes/iOS/6236191244602421).

**Comment by brajes** [comment]
Hi @yyzhu 
Thank very much for the updates.
